### ISSUE TITLE
Create Releases for MacOS, Windows and Linux

### ISSUE THREAD

**Issue by deckarep** [deckarep/larry-pops-up#1]
I need to figure out an effective method of bundling this across the 3 different operating systems.

/cc @doomlazer - thought you might get a kick out of this...I threw this together in about an an hour...there's some fun audio clips you might want to use.

**Comment by Doomlazer** [comment]
Looks interesting but I can't get zig to compile it, so I'll have to wait for the binaries to check it out.

**Comment by deckarep** [comment]
Yeah, i'm going down the path of trying to get some build artifacts but having lots of trouble. I've never done this with Zig but apparently the build system and cross-compilation magic is very innovative. I'll get there.

**Comment by deckarep** [comment]
@Doomlazer - if you're curious: 
https://github.com/deckarep/larry-pops-up/actions/runs/10294264765/artifacts/1788334640

There's my first successful build artifact but it will only run on an *intel* macos, if you have an M{1,2,3} it won't work. I have to figure out how to build for the M chips next.

Just download it, unzip it, chmod +x [file] to make it runnable. And since it's not a signed binary you'll have to right click to open and give it privileges to run.

I tested it and it works as a single binary...all the assets are embedded within the binary so it doesn't require any further dependencies which is nice.

Anyhow :) - let me know if you try it...no pressure though.

**Comment by Doomlazer** [comment]
It does work on my intel mac. If I add .app to the file it says it's only compatible with 14.5. My 2014 laptop can't even get to os 12, but I wonder if it would work on a Sonoma machine without needing to chmod?

It is a pretty cool little lost app, but in its current form it's a bit difficult to "experience". It would be nice if this was a bit more user friendly to launch, the average mac user would be afraid to chmod a file which limits the audience. Also, it's going to be difficult to replicate the "pop-up" aspect of the original program. I guess you could create a cron job to trigger it at a set interval, but again that's too much for the average user. Maybe asking for an interval at launch and then sleeping between WAVs would be closer to the original experience?

 It would also be nice to support the user supplied wavs somehow; the cyberlarry 2000 jokes would work well with this.

Have you considered an electron app? I haven't really looked into it, but it might skirt some of the cross platform compatibility issues?

**Comment by deckarep** [comment]
Thanks for trying it...I'll admit of all the possible Sierra stuff out there this one is likely one of the most absolute useless things to try.

The intention of having you try it out was to just simply see if it would launch and show the image, play the .wav files. It literally just chooses one random image, and one random wav file, plays it then quits.

In order to do this "right" though for real everyday users: I would need to bundle it properly as a Mac App bundle, give it a thumbnail, add a launch configuration which would allow me to run it as a system level service on behalf of the user.

And then like you said, provide the user with some configuration options (easily accessible) so they could install and run it and/or customize it's behavior. Such as adjust how often it pops up, or specify their own folder of assets, etc.

I haven't done any of that...and then I'd have to do something similar for Windows and eventually Linux. The reason I chose Zig is because I've been working with this language now for the last couple years for personal projects and it's pretty much as low-level as it gets because it's considered as a modern C replacement.

But even choosing Zig has it's warts because it's not even a 1.0 level language.

I could go down the route of an Electron app and bundle it in a web-form and probably have an easier time with installs and launching the software but I'm so in love with Zig that all of these throw away projects are just an excuse for me to practice in a language which I have a feeling will explode in popularity in a few years time.

That's my rationale at least...but you are definitely correct that the app has not yet been production-ized for end users by any means but it shouldn't be terribly hard to get it there.

**Comment by Doomlazer** [comment]
It's interesting because it wouldn't be too difficult to really flesh out, but at the same time it's a novelty that no sane person would want running on their computer very long. Even back in the day, I find it difficult to believe anyone ever left this active for more than 48 hours max.

Still, it is fun to check out and the WAVs that I've heard are pretty funny. Making this run-able on modern systems is definitely an improvement to it's current state as a forgotten spin off people scroll by while browsing Al's site.

Looking at the Zig introduction, the language seems interesting. I think I'd need to dig a bit deeper to really understand where its shines though.
